pkg/webhook/csidriver/csidriver.go

```diff
@@ -81,6 +101,58 @@ func (s *SharedResourcesCSIDriverWebhook) authorizePod(request admissionctl.Requ
 	return ret
 }
 
+func (s *SharedResourcesCSIDriverWebhook) authorizeSharedSecret(request admissionctl.Request) admissionctl.Response {
+	klog.V(2).Info("admitting shared secret with SharedResourceCSIDiver")
+
+	SharedSecret, err := s.renderSharedSecret(request)
+	if err != nil {
+		klog.Error(err, "Couldn't render a shared secret from the incoming request")
+		return admissionctl.Errored(http.StatusBadRequest, err)
+	}
+
+	return s.authorizeResource(request, SharedSecret.ObjectMeta.Name)
+}
+
+func (s *SharedResourcesCSIDriverWebhook) authorizeSharedConfigMap(request admissionctl.Request) admissionctl.Response {
+	klog.V(2).Info("admitting shared configmap with SharedResourceCSIDiver")
+	SharedConfigMap, err := s.renderSharedConfigMap(request)
+	if err != nil {
+		klog.Error(err, "Couldn't render a shared configmap from the incoming request")
+		return admissionctl.Errored(http.StatusBadRequest, err)
+	}
+
+	return s.authorizeResource(request, SharedConfigMap.ObjectMeta.Name)
+}
+
+// authorizeResource checks whether the shared resources can be created
+// with "openshift-" prefix or not.
+func (s *SharedResourcesCSIDriverWebhook) authorizeResource(request admissionctl.Request, resource string) admissionctl.Response {
+
+	var (
+		ret           admissionctl.Response
+		allowResource bool
+	)
+
+	if strings.HasPrefix(resource, "openshift-") {
+		for key, _ := range config.LoadedConfig.AuthorizedSharedResources {
+			if strings.EqualFold(key, resource) {
+				allowResource = true
+				break
+			}
+		}
+		if !allowResource {
+			ret = admissionctl.Denied("Shared resource with prefix 'openshift-' is not allowed, unless listed under OCP pre-populated shared resource")
+			ret.UID = request.AdmissionRequest.UID
+			return ret
+		}
```

Comment:
```suggestion
		if _, ok := config.LoadedConfig.AuthorizedSharedResources[resource]; !ok {
			ret = admissionctl.Denied("Shared resource with prefix 'openshift-' is not allowed, unless listed under OCP pre-populated shared resource")
			ret.UID = request.AdmissionRequest.UID
			return ret
		}
```

Reply:
You only need to check if the resource exists in the map, you don't need to iterate over the map like you would for a slice.